### PR TITLE
Automatically generate Keywoards Documentation using libdoc with CI workflow

### PR DESCRIPTION
**Pull request by WiktorG351**

### CODE REVIEW

File: .github/workflows/auto-docs.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        python3 -m venv robot-venv
+        source robot-venv/bin/activate
+        pip install -r requirements.txt
```
Comment (artur-rs) [inline .github/workflows/auto-docs.yml:25]: @WiktorG351 I believe only the last command is necessary

File: .github/workflows/auto-docs.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    - name: Generate documentation
+      run: |
+        chmod +x scripts/create-docs.sh
+        scripts/create-docs.sh
```
Comment (artur-rs) [inline .github/workflows/auto-docs.yml:30]: If we push the file attributes once, and do not change it, we do not need to chmod it everytime
```suggestion
      run: cd ${GITHUB_WORKSPACE}/scripts && bash create-docs.sh
```

File: .github/workflows/auto-docs.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    - name: Deploy to GitHub Pages
+      uses: peaceiris/actions-gh-pages@v3
```
Comment (artur-rs) [inline .github/workflows/auto-docs.yml:33]: v4?  https://github.com/peaceiris/actions-gh-pages

File: README.md [hunk 4]
```diff
@@ -224,6 +225,38 @@ found in 'open-source-firmware-validation' folder.
 * [Renaming Test Cases](https://robotidy.readthedocs.io/en/stable/transformers/RenameTestCases.html)
 * [Renaming Variables](https://robotidy.readthedocs.io/en/stable/transformers/RenameVariables.html)
 
+## Generating documentation
```
Comment (artur-rs) [inline README.md:228]: @WiktorG351, let's update this section after the successful integration of the gh pages

File: scripts/create-docs.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+find lib -name "*.robot" -exec cat {} + >> "$TEMP_DIR/all-keywords.robot"
+
+python3 robot-venv/bin/libdoc "$TEMP_DIR/all-keywords.robot" "$TEMP_DIR/all-keywords.html" >/dev/null 2>&1
+
+if [ $? -ne 0 ]; then
+  echo "libdoc command failed"
+  exit 1
+fi
+
+cp "$TEMP_DIR/all-keywords.html" .
```
Comment (artur-rs) [inline scripts/create-docs.sh:15]: @WiktorG351 where is `docs` directory used? yml has it specified

**Review by artur-rs — CHANGES_REQUESTED** [review]

File: scripts/create-docs.sh (hunk 5 above)
Comment (WiktorG351) [reply scripts/create-docs.sh]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-aa664a4f27e008df9a11dea5effa5d1adfa409490475f026c53ca44419606b81L15-R15)

File: .github/workflows/auto-docs.yml (hunk 3 above)
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L33-R29)

File: .github/workflows/auto-docs.yml (hunk 2 above)
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L28-R26)

File: .github/workflows/auto-docs.yml (hunk 1 above)
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L22-R23)

File: .github/workflows/auto-docs.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+jobs:
+  generate-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
```
[the remark this answers is not in the document]
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L17-R18)

File: .github/workflows/auto-docs.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    branches:
+      - master
+
+jobs:
+  generate-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml]: @artur-rs adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L14-R15)

File: .github/workflows/auto-docs.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
+name: Auto Docs
+
+on:
+  push:
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (WiktorG351) [reply .github/workflows/auto-docs.yml:6]: @artur-rs added [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/a2bc617ebb04f86174ec5ae2e46003f1aebd3a4f#diff-ad22275c389220855fb5c424fe7960b59b03afdc63661e3dac08f4bab54c9d45L6-R7)

File: README.md (hunk 4 above)
Comment (WiktorG351) [reply README.md:228]: @artur-rs added [here](https://github.com/Dasharo/open-source-firmware-validation/pull/352/commits/0e9c71469d6f983f1c8c0bd884634526ba577e2d)

**Review by artur-rs — APPROVED** [review]

**Pull request merged**